Support block comment

jamesreggio opened the pull request:
This PR supersedes #8, which has been dormant for quite some time.

Instead of requiring a `--block` argument or looking at the file type, we merely check the type of comment already present in the stream and use that to insert the new, relative `sourceMappingURL`.

(This PR expects #12 to be merged first.)

thlorenz commented on .gitignore:
```diff
@@ -10,6 +10,7 @@ lib-cov
 pids
 logs
 bundle.*
+to.*
```
Why is this here?
You added a `to.css` anyways.

Please remove, just select which files to add, i.e. don't `git add .`.

jamesreggio replied:
Sorry, I have no idea how this got here. I'll clean it up after we resolve #12, since these two branches are going to be in conflict until then.

thlorenz commented:
LGTM except for the NIT.

thlorenz commented:
OK gonna wait til the other PR gets merged and then re-evaluate this one after you rebased on the merge.

jamesreggio commented:
Alright @thlorenz, I rebased this against my other changes in #12 and squashed again.

I remember now why I added `to.*` to the `.gitignore`. It's consistent with the `bundle.*` [line](https://github.com/jamesreggio/exorcist/blob/block-comment/.gitignore#L12) above it, which ignores the `bundle.js.map` file [emitted by the tests](https://github.com/jamesreggio/exorcist/blob/block-comment/test/exorcist.js#L10-11).

If you're still bothered by it, do you suggest I remove both and clean up those files at the time the tests terminate?

thlorenz commented:
> do you suggest I remove both and clean up those files at the time the tests terminate

Yes please do. Tests should always clean up after themselves. You can register the clean up function with `t.on('end'` of the test that creates the files.
Once that's done remove the `.gitignore` entry.

It seems like I haven't been purely following this myself as I only clean up _before_ [each test run](https://github.com/jamesreggio/exorcist/blob/block-comment/test/exorcist.js#L16-L19).
I may have kept these around for inspection during development and forgotten to fix that.

If you want you could fix that as well otherwise I'll take care of that once we merge this PR.


Thanks.

jamesreggio commented:
Alright @thlorenz, I added the map-file cleanup to the tests. Could you re-review and merge?

(Also, I think this PR satisfies the functionality in #8 that you wanted to support—you explicitly didn't believe in the change to have `exorcist` create missing directories.)

thlorenz commented:
Thanks this looks very good now.
Will review in detail on Monday and merge.
Please ping me if I forget.

thlorenz commented:
Thanks merged 0b7b980eafcb963b99884cb9649ffab9b0710965 and published as `0.3.0`
